kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -4520,6 +4525,18 @@ protected void removeResource(ContentResourceEdit edit, boolean removeContent) t
 		// htripath -store the metadata information into a delete table
 		// assumed uuid is not null as checkExplicitLock(id) throws exception when null
 
+		//check if resourec is of type CitationList and clean up citation tables
+		if(removeContent){
+			if(edit.getResourceType().equals(CitationService.CITATION_LIST_ID)){
+				try{
+					log.info("Removing CitationList: " + new String(edit.getContent()));
+					citationService.hardDeleteCitation(new String(edit.getContent()));
+				}catch (ServerOverloadException e){
+					log.error("Error removing CitationList " + e);
+				}
```

Comment:
```suggestion
				String data = new String(edit.getContent());
				try {
					log.debug("removing citation list [{}]", data);
					citationService.hardDeleteCitation(data);
				} catch (ServerOverloadException e) {
					log.warn("Attempting to remove citation list [{}], {}", data, e);
				}
```

Reply:
edit.getContent() does throw ServerOverloadException and can't be moves outsite of try block